app/controller/button/StreetViewTool.js

```diff
@@ -0,0 +1,380 @@
+                var win = me.streetViewWin;
+                var title = view.svWinTitlePrefix + ' Image Date: ' + data.imageDate;
+
+                win.setTitle(title);
+                win.show();
+
+                // create new Street View panorama
+                var panoConf = {
+                    position: latLng,
+                    pov: view.svDefaultPov
+                };
+                me.svPanorama = new google.maps.StreetViewPanorama(me.svDiv);
+                me.svPanorama.setPov(panoConf.pov);
+                me.svPanorama.setPosition(panoConf.position);
+
+                // initially draw the position on the map and set heading
+                me.drawPositionFeature(me.gmapsLatLng2olCoord(latLng));
+                me.updatePositionFeature();
+
+                google.maps.event.addListener(
+                    me.svPanorama, 'pov_changed', me.handlePovChanged.bind(me));
+                google.maps.event.addListener(
+                    me.svPanorama, 'position_changed', me.handlePositionChanged.bind(me));
+
+            } else {
+                // inform subscribers that there was no panorama for clicked pos
+                me.getView().fireEvent('missingpanorama', {
+                    msg: 'No StreetView panorama available at clicked location'
+                });
+            }
+        });
+    },
+
+    /**
+     * Draws the position feature on the map. Clears an eventually existing
+     * position feature before adding the new one.
+     *
+     * @param  {ol.Coordinate} coord Position to draw on the map
+     * @private
+     */
+    drawPositionFeature: function (coord) {
+        var me = this;
+        var vectorSource = me.vectorLayer.getSource();
+
+        // remove existing position feature
+        vectorSource.clear();
+        // add new position feature
+        var posFeat = new ol.Feature({
+            geometry: new ol.geom.Point(coord),
+        });
+        vectorSource.addFeature(posFeat);
+    },
+
+    /**
+     * Updates an existing position feature by applying the current position and
+     * heading of the currently opened SV panorama.
+     */
+    updatePositionFeature: function () {
+        var me = this;
+        var newHeading = me.getHeadingRad();
+        var newPosCoord = me.getPositionCoord();
+        var vectorSource = me.vectorLayer.getSource();
+        var feat = vectorSource.getFeatures()[0];
+
+        if (feat) {
+            // move position feature to current position
+            feat.getGeometry().setCoordinates(newPosCoord);
+
+            // rotate position feature to current heading
+            me.vectorLayer.getStyle().getImage().setRotation(newHeading);
+            vectorSource.refresh();
+        }
+    },
+
+    /**
+     * Handles 'pov_changed' event of the SV panorama.
+     * Triggers #updatePositionFeature.
+     *
+     * @private
+     */
+    handlePovChanged: function () {
+        var me = this;
+        me.updatePositionFeature();
```

Comment:
Aliasing unneeded here IMO. Also below.

Reply:
I suggest to leave this here beacuse of backward compatibility to the old version and more important
because it gives us the envelope for further things to be done on the certain events in the future.